pkg/services/baremetal/host/host.go

```diff
@@ -1789,65 +1804,222 @@ func analyzeSSHOutputProvisioned(out sshclient.Output) (isTimeout, isConnectionR
 // next: Stays in Provisioned (final state)
 func (s *Service) actionProvisioned(ctx context.Context) actionResult {
 	// set host to provisioned
-	conditions.MarkTrue(s.scope.HetznerBareMetalHost, infrav1.ProvisionSucceededCondition)
 
 	rebootDesired := s.scope.HetznerBareMetalHost.HasRebootAnnotation()
-	isRebooted := s.scope.HetznerBareMetalHost.Spec.Status.Rebooted
-	creds := sshclient.CredentialsFromSecret(s.scope.OSSSHSecret, s.scope.HetznerBareMetalHost.Spec.Status.SSHSpec.SecretRef)
-	in := sshclient.Input{
-		PrivateKey: creds.PrivateKey,
-		Port:       s.scope.HetznerBareMetalHost.Spec.Status.SSHSpec.PortAfterInstallImage,
-		IP:         s.scope.HetznerBareMetalHost.Spec.Status.GetIPAddress(),
+
+	host := s.scope.HetznerBareMetalHost
+
+	if !rebootDesired {
+		host.Spec.Status.Rebooted = false
+		host.Spec.Status.ExternalIDs.RebootAnnotationNodeBootID = ""
+		host.Spec.Status.ExternalIDs.RebootAnnotationSince.Time = time.Time{}
+		conditions.MarkTrue(s.scope.HetznerBareMetalHost, infrav1.ProvisionSucceededCondition)
+		return actionComplete{} // Stays in Provisioned (final state)
 	}
-	sshClient := s.scope.SSHClientFactory.NewClient(in)
 
-	if rebootDesired {
-		if isRebooted {
-			// Reboot has been done already. Check whether it has been successful
-			// Check hostname with sshClient
-			out := sshClient.GetHostName()
+	if host.Spec.Status.ExternalIDs.RebootAnnotationSince.IsZero() {
+		host.Spec.Status.ExternalIDs.RebootAnnotationSince = metav1.Now()
+	}
 
-			wantHostName := s.scope.Hostname()
+	wlClient, err := s.scope.WorkloadClusterClientFactory.NewWorkloadClient(ctx)
+	if err != nil {
+		err = fmt.Errorf("actionProvisioned (Reboot via Annotation), failed to get wlClient: %w", err)
+		conditions.MarkFalse(host, infrav1.ProvisionSucceededCondition,
+			infrav1.StillProvisioningReason,
+			clusterv1.ConditionSeverityWarning, "%s",
+			err.Error())
+		return actionError{err: err}
+	}
 
-			if trimLineBreak(out.StdOut) == wantHostName {
-				// Reboot has been successful
-				s.scope.HetznerBareMetalHost.Spec.Status.Rebooted = false
-				s.scope.HetznerBareMetalHost.ClearRebootAnnotations()
+	machine, err := util.GetOwnerMachine(ctx, s.scope.Client, s.scope.HetznerBareMetalMachine.ObjectMeta)
+	if err != nil {
+		err = fmt.Errorf("actionProvisioned (Reboot via Annotation), GetOwnerMachine failed: %w",
+			err)
+		conditions.MarkFalse(host, infrav1.ProvisionSucceededCondition,
```

Comment:
We can skip this condition. We also don't set a condition if another object is not found (e.g. HetznerBareMetalMachine)

Reply:
done.